File: frontend/src/services/reservationService.js
import apiClient from "../api/axios";
import API_ENDPOINTS from "../constants/apiEndpoints";

const reservationService = {
  /**
   * Get all reservations
   */
  getReservations: async () => {
    const response = await apiClient.get(
      API_ENDPOINTS.RESERVATIONS
    );

    return response.data;
  },

  /**
   * Get reservations by customer email
   */
  getReservationsByCustomerEmail: async (
    customerEmail
  ) => {
    const response = await apiClient.get(
      `${API_ENDPOINTS.RESERVATIONS_BY_CUSTOMER}/${encodeURIComponent(
        customerEmail
      )}`
    );

    return response.data;
  },

  /**
   * Create reservation
   */
  createReservation: async (
    reservationRequest
  ) => {
    const response = await apiClient.post(
      API_ENDPOINTS.RESERVATIONS,
      reservationRequest
    );

    return response.data;
  },

  /**
   * Update reservation
   */
updateReservation: async (
  reservationId,
  request
) => {
  const response = await apiClient.put(
    `${API_ENDPOINTS.RESERVATIONS}/${reservationId}`,
    request
  );

  return response.data;
},

  /**
   * Cancel reservation
   */
  cancelReservation: async (id) => {
    await apiClient.delete(
      `${API_ENDPOINTS.RESERVATIONS}/${id}`
    );
  },
};

export default reservationService;

File: backend/src/main/java/com/ajay/carrental/service/ReservationService.java
package com.ajay.carrental.service;

import com.ajay.carrental.dto.request.ReservationRequest;
import com.ajay.carrental.dto.request.ReservationUpdateRequest;
import com.ajay.carrental.dto.response.ReservationResponse;
import com.ajay.carrental.entity.Customer;

import java.util.List;
import java.util.UUID;

public interface ReservationService {

    ReservationResponse reserve(ReservationRequest request);

    ReservationResponse modify(UUID reservationId, ReservationUpdateRequest request);

    void cancel(UUID reservationId);

    List<ReservationResponse> getAllReservations();

    List<ReservationResponse> getReservationsByCustomerEmail(String email);
}